BlockServer/core/ioc_control.py

```diff
@@ -33,39 +33,38 @@ def __init__(self, prefix):
         """
         self._proc = ProcServWrapper(prefix)
 
-    def start_ioc(self, ioc):
-        """Start an IOC.
-
+    def _perform_ioc_control(self, control_function, ioc):
+        """Performs an ioc control operation.
         Args:
-            ioc (string): The name of the IOC
+            control_function: A reference to the function to perform.
+            ioc (str): The ioc to perform the function.
         """
         try:
-            self._proc.start_ioc(ioc)
+            control_function(ioc)
             if ioc != "ALARM":
                 AlarmConfigLoader.restart_alarm_server(self)
         except Exception as err:
-            print_and_log("Could not start IOC %s: %s" % (ioc, str(err)), "MAJOR")
+            print_and_log("Could not perform {} IOC {}: {}".format(control_function.__name__, ioc, str(err)), "MAJOR")
+
+    def start_ioc(self, ioc):
+        """Start an IOC.
+
+        Args:
+            ioc (string): The name of the IOC
+        """
+        self._perform_ioc_control(self._proc.start_ioc, ioc)
 
     def restart_ioc(self, ioc, force=False):
         """Restart an IOC.
 
-        Note: restarting an IOC automatically sets the IOC to auto-restart, so it is neccessary to reapply the
-        previous auto-restart setting
-
         Args:
             ioc (string): The name of the IOC
             force (bool): Force it to restart even if it is an IOC not to stop
         """
         # Check it is okay to stop it
         if not force and ioc.startswith(IOCS_NOT_TO_STOP):
             return
-        try:
-            auto = self._proc.get_autorestart(ioc)
```

Comment:
Note that my refactoring removed this line, which was the one originally failing the ticket.